### ISSUE TITLE
nav_translations do not work with awesome-pages

### ISSUE THREAD

**Issue by phasel74** [ultrabug/mkdocs-static-i18n#65]
The `mkdocs.yml`
```YAML
site_name: Test
nav:
  - Home: index.md
theme:
  name: material
plugins:
  - awesome-pages
  - i18n:
      default_language: en
      languages:
        en: English
        de: Deutsch        
      nav_translations:
        de:
          'Home': Startseite
```

with `index.md`
```Markdown
# Welcome!
```

and `index.de.md`
```Markdown
# Herzlich Willkommen!
```
gives the German website

![grafik](https://user-images.githubusercontent.com/48174934/140091752-727fc2c5-5fa9-442e-b01d-f7050c1dd20d.png)

When I remove awesome-pages I get the expected website

![grafik](https://user-images.githubusercontent.com/48174934/140092149-498c2311-6622-4dbc-b649-8656dea08a28.png)

Is this an issue of awesome-pages or of i18n?

**Comment by ultrabug** [comment]
Hi @phasel74 ; when building the site, you should see a log like this:

```
INFO     -  Translating navigation to de
```

which means that the mkdocs-static-i18n plugin has done its job. But then I guess that the awesome-pages plugin comes into play and somehow overrides the work done by i18n...

**Comment by ultrabug** [comment]
```
INFO     -  Ignoring 'nav_translations' option: the 'awesome-pages' plugin breaks the ability to translate navigation titles
```

An informative message will now tell you about this, thanks for reporting!

**Comment by scmcca** [comment]
I'm dealing with the same issue. Are there any work arounds or planned updates to fix this?

**Comment by ultrabug** [comment]
Well as usual in this sort of inter compatibility problem @scmcca there are two sides on the same problem and while this issue has been reported here, maybe it should have been reporter on the mkdocs-awesome-pages-plugin.

From my understanding, the i18n plugin here modifies the files structure directly while the awesome-pages one recreates it to form a custom Navigation object in its internals which breaks what i18n has done to translate navigations.

I'm issuing a friendly ping to @lukasgeiter in case he can step in and help on this one.

**Comment by lukasgeiter** [comment]
I'm pretty sure I know what's happening. Here are the relevant details from the inner workings of both plugins:

### i18n

The i18n plugin triggers the `nav` event a second time in it's `on_post_build` method:

https://github.com/ultrabug/mkdocs-static-i18n/blob/d4362481f8cbea055597c230e02b95d45c5cbff3/mkdocs_static_i18n/plugin.py#L658-L661

However, during this second run the plugin also skips translating the navigation, because `files.translated` is `True`:

https://github.com/ultrabug/mkdocs-static-i18n/blob/d4362481f8cbea055597c230e02b95d45c5cbff3/mkdocs_static_i18n/plugin.py#L560-L564

### awesome-pages

In it's `on_nav` method, awesome-pages ignores the passed in `nav` object and instead looks at the `nav` from the `mkdocs.yml` config. It does this to work around the validation that mkdocs applies to the `nav` config. Some of the plugin's features would cause mkdocs to report warnings about invalid `nav` config entries. To avoid that, the plugin does this:

1. In `on_config`:
    1. Make a copy of `config['nav']`
    2. Set `config['nav']` to `None` so mkdocs doesn't read the "invalid" entries
2. In `on_nav`:
    1. Discard the passed in `nav` because it was generated without `config['nav']`
    2. Restore copy of `config['nav']`
    3. Build real navigation

Under normal circumstances this works fine if the plugin is listed before other plugins that modify the navigation. But because i18n triggers the `nav` event a second time, this causes problems.

---

Side note: The described workaround actually only needs to happen if the `mkdocs.yml` contains any of these "invalid" entries. However, I just discovered a bug that causes the workaround to always be executed. I'm going to fix that and post an update here once the version is out.

---

### Conclusion

I'm not sure what the best way is to fix this compatibility issue. @ultrabug would it be possible to remove the code that triggers the `nav` event a second time? As far as I can tell, that would make the two plugins 100% compatible, as long as awesome-pages is put before i18n in the list of plugins.

**Comment by ultrabug** [comment]
Thanks for your time and analysis @lukasgeiter ; this gives me hope that we can find a solution.

> would it be possible to remove the code that triggers the nav event a second time? 

I'm not so sure yet since the i18n plugin has to trigger a build for every language version, each of the versions having their own files & navigation... I will consider and try to create those earlier instead.

The irony is that I did add the on_nav for compatibility with your plugin :D

> as long as awesome-pages is put before i18n in the list of plugins.

Do you actually check for this? or issue a warning otherwise on your part? because this is something that could be checked, reported or even fixed on the plugin's side if I'm not mistaken.

**Comment by ultrabug** [comment]
Quick follow-up so far, I tried:

- not calling `on_nav` a second time, no effect
- awesome-pages before or after i18n has no effect either
- building `nav` directly on `on_nav` and not calling on_nav, no effect

I'll spend an hour more on this tonight if I can, so far, none of the above ideas work.

**Comment by ultrabug** [comment]
@lukasgeiter I just went back to check on this issue and I can say that since your 2.6.1 fix, the translation works as expected and our plugins are fully compatible ;)

**Comment by lukasgeiter** [comment]
@ultrabug I believe that's only true if the `nav` option in `mkdocs.yml` doesn't contain any `...` entries.

**Comment by ultrabug** [comment]
ok @lukasgeiter let me check that then

**Comment by ultrabug** [comment]
@lukasgeiter 

```
nav:
  - topic2
  - ...
  - topic1
  ```
  
  works as intended

![2021-12-02-231952_630x224_scrot](https://user-images.githubusercontent.com/474389/144512448-f2b62e32-4aa1-4c0a-a583-05b8cfbdbd04.png)

**Comment by lukasgeiter** [comment]
@ultrabug Yes, but what if you put the same thing in `mkdocs.yml`?

**Comment by ultrabug** [comment]
hi, #78 will be merged and makes sure that :

- awesome-pages is always called first on events
- on_nav is not called recursively on all plugins for each language, so it is only called once from now on

**Comment by ultrabug** [comment]
> @ultrabug Yes, but what if you put the same thing in mkdocs.yml?

looks like it behaves badly, even with #78 which I don't really understand why... I don't see why awesome-pages behaves differently between a .pages or the mkdocs.yml nav tbh

**Comment by lukasgeiter** [comment]
@ultrabug Because `mkdocs.yml` cannot contain things that MkDocs doesn't support (like `...` entries). As a workaround, the plugin removes `nav` from the config object as described in [this comment](https://github.com/ultrabug/mkdocs-static-i18n/issues/65#issuecomment-974352259).

**Comment by ultrabug** [comment]
@lukasgeiter I got that yes, I just don't see how I can make something more on my side I'm sorry

**Comment by lukasgeiter** [comment]
@ultrabug Me neither, unfortunately.

**Comment by ultrabug** [comment]
I'll close here then, I believe we've done better than we hoped for already

This will be released with 0.30 FYI @phasel74
